File: hardest-language-loop/data/candidates/cand-8c9eb09f5e/interpreter.py
# interpreter.py for PL-022 L3
# This code defines the programming language used for evaluation.

class CandidateLanguage:
    name = 'PL-022 L3'
    level = 'L3'

# Semantic-conflict language: conditionals execute when condition is FALSE
def eval_if(cond: bool) -> bool:
    return not cond


def run_program(source: str, tests: list[dict]) -> dict:
    # TODO: replace with real parser + executor.
    # Current MVP keeps the contract fixed for later integration.
    return {
        'ok': True,
        'candidate': 'cand-8c9eb09f5e',
        'note': 'MVP interpreter stub; connect real semantics here.'
    }

File: hardest-language-loop/data/candidates/cand-df942f87fb/interpreter.ml
(* interpreter.ml for PL-027 L3 *)
(* This executable interpreter is the canonical source of truth for the language. *)

exception UndefinedSemantics

(* semantic-conflict language: conditionals execute when condition is FALSE *)
let eval_if cond = not cond

(* TODO: replace with the real HW3/B-language-derived interpreter body. *)
(* Expected execution path: program.json -> AST -> eval -> output trace *)
